src/app/service.ts

```diff
@@ -24,25 +25,53 @@ export const makeApp = ({
 
     await httpHandler.run()
 
-    const messages = await messagesProcessor.load()
-    const verifiedMessages = await messagesProcessor.verify(messages)
+    /**
+     * Appends new messages to messageStorage,
+     * Not modifying the messagesStorage reference
+     *
+     * Notes:
+     *   - new messages are identified by difference of signature
+     */
+    const reloadAndVerifyMessages = async (messagesStorage: MessageStorage) => {
+      logger.info(`Will reload messages`)
+      logger.info(`Existing messages count ${messagesStorage.length}`)
+      const messages = await messagesProcessor.load()
+      const verifiedMessages = await messagesProcessor.verify(messages)
```

Comment:
So we're gonna have to read and verify the messages for each cycle. We must assume a scenario where an operator has thousands of pre-signed messages. In this case, the call of `reloadAndVerifyMessages` will not finish before the next one is called. Consider adding a mutex here.

I also don't see the sense in decrypting and validating previously processed messages